### ISSUE TITLE
Performance degradation after 1.0.2

### ISSUE THREAD

**Issue by hpresnall** [logstash-plugins/logstash-input-lumberjack#49]
I recently upgraded to Logstash 1.5.4 (from 1.5.2) and am seeing a pretty large performance regression when using the lumberjack input plugin.

With a simplified configuration using just the input plugin, and no output plugins or filters, 1.5.2 was able to achieve over 20,000 events per second on a test vm. With 1.5.4, this drops to about 11,000 events per second. I can also reproduce this on 1.5.2 by upgrading the lumberjack plugin to the same 1.0.5 version that works with 1.5.4. So, this appears to be related only to the plugin and not base logstash.

Digging into the changes made, I was able to fix almost all of this regression by changing `  BUFFERED_QUEUE_SIZE` in lumberjack.rb back to 20. It was changed due to issue #43 where there are comments about why this was changed to 1. It appears that this was not beneficial but I am not sure if there's a better way to handle this.

**Comment by ph** [comment]
I'll do a bit more test on my side if I can enable it back, but in reality this buffer should not exist in the first place and the main SizeQueue should support push with timeout out of the box. The problem is this double queue artificially boost what logstash queues can actually contains, but having a really low value is probably making the lock a bit active.

**Comment by ph** [comment]
The problem in reality is to make sure we correctly tell the LSF to actually resend his payload so we don't lose evenst in transit.

**Comment by ph** [comment]
@hpresnall I agree this is an issue, I'll do a test on core to add the timeout directly on the main sizequeue class and remove the internal buffer. When I have released the updated lumberjack we did not want to change the semantic of the queue, which would require a new release.

**Comment by hpresnall** [comment]
Great. I can test an update when it's available.

**Comment by ph** [comment]
it may take some time, but keep in mind adding a bigger value on the buffered size queue can lead to the lost of events. Since we could already have sent the `ACK` for the LSF before having a timeout exception, the data lost should be in all cases small but the risk is there.
